time_tracker_api/time_entries/time_entries_model.py

```diff
@@ -116,21 +125,30 @@ def find_all(
         conditions: dict = {},
         date_range: dict = {},
     ):
-        custom_sql_conditions = []
-        custom_sql_conditions.append(
-            self.create_sql_date_range_filter(date_range)
-        )
+        custom_sql_conditions = [self.create_sql_date_range_filter(date_range)]
 
-        custom_params = self.generate_params(date_range)
+        if event_context.is_admin:
+            conditions.pop("owner_id")
 
-        return CosmosDBRepository.find_all(
+        custom_params = self.generate_params(date_range)
+        time_entries = CosmosDBRepository.find_all(
             self,
             event_context=event_context,
             conditions=conditions,
             custom_sql_conditions=custom_sql_conditions,
             custom_params=custom_params,
         )
 
+        if time_entries:
```

Comment:
Can we move logic from lines 142-148  to an external function ? I think we can apply here https://refactoring.guru/extract-method . That function could go in `time_entries/custom_modules/utils` in a similar way as I did with function `add_project_name_to_time_entries` . The new function could be called `projects_available_for_current_time_entries`